DfciPkg/SettingsManager/SettingsManagerProvider.c

```diff
@@ -405,23 +405,27 @@ ProviderValueAsAscii (
         break;
       }
 
-      Value = AllocateZeroPool (SECURE_BOOT_ENUM_STRING_SIZE);
-      if (Value == NULL) {
-        DEBUG ((DEBUG_ERROR, "Failed - Couldn't allocate for string. \n"));
-        break;
-      }
-
       if (b == 0) {
-        AsciiStrCpyS (Value, SECURE_BOOT_ENUM_STRING_SIZE, "MsOnly");
+        AsciiString = "MsOnly";
       } else if (b == 1) {
-        AsciiStrCpyS (Value, SECURE_BOOT_ENUM_STRING_SIZE, "MsPlus3rdParty");
+        AsciiString = "MsPlus3rdParty";
       } else if (b == 3) {
         // This is a special case.  Only supported as output.
-        AsciiStrCpyS (Value, SECURE_BOOT_ENUM_STRING_SIZE, "Custom");
+        AsciiString = "Custom";
       } else {
-        AsciiStrCpyS (Value, SECURE_BOOT_ENUM_STRING_SIZE, "None");
+        AsciiString = "None";
+      }
+
+      ValueSize = AsciiStrLen (AsciiString) + 1;

```

Comment:
nit: should probably use SafeStrings, even though this is not user input. 
```suggestion
      ValueSize = AsciiStrnSizeS (AsciiString, DFCI_MAX_ID_LEN) + 1;
```